network/src/services/outbound_peer.rs

```diff
@@ -70,7 +55,7 @@ impl OutboundPeerService {
         let p2p_control = self.p2p_control.clone();
         for (peer_id, addr) in peers
             .into_iter()
-            .filter(|(peer_id, _addr)| self.network_state.local_peer_id() != peer_id)
+            .filter(|(peer_id, addr)| self.network_state.can_dial(peer_id, addr))
```

Comment:
duplicate filter? can_dial is already called in dial method

Reply:
dial may called elsewhere